lib/spack/spack/build_types.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""Build types can add different flags depending on the compiler.
+"""
+
+
+class BuildTypeBase(object):
+    """A build type base provides base functions to look up flags for a compiler.
+    """
+    default = None
+
+    def __init__(self, spec):
+        self.spec = spec
+        self.default = []
+
+    @property
+    def name(self):
+        """get the name for the class
+        """
+        return self.__class__.__name__
```

Comment:
This doesn't match how you're using it in the packages above. We need to translate it to the lower-case version.

Reply:
What was the function you were talking about to list all the classes? The list_classes in the llnl util requires a folder to parse.

Reply:
Ahh this one.
```
mod_to_class("basic_debug")
Out[11]: 'BasicDebug'
```
but I guess we need the reverse.

Reply:
I thought there was an inverse already, but apparently not. I guess we can write that, it's not too complicated. There'll be a slightly weird regex to get the separate names.

Reply:
https://github.com/spack/spack/pull/23231/commits/1b39415174955f44d26ea45c893666029d8f5c69